page-tracking/src/collections/page_arc.rs

```diff
@@ -32,20 +33,41 @@ struct PageArcInner<T> {
 pub struct PageArc<T>(NonNull<PageArcInner<T>>);
 
 impl<T> PageArc<T> {
+    /// Try to create a `PageArc` that wraps the given data using
+    /// `pages` to store it, returning it to its previous owner on
+    /// `drop()` using `page_tracker`. Returns `None` if not enough
+    /// pages are supplied.
+    pub fn try_new(
+        data: T,
+        pages: SequentialPages<InternalClean>,
+        page_tracker: PageTracker,
+    ) -> Option<Self> {
+        if core::mem::size_of::<PageArcInner<T>>() <= pages.length_bytes() as usize {
+            Some(Self::new_with(data, pages, page_tracker))
+        } else {
+            None
+        }
+    }
+
     /// Creates a `PageArc` that wraps the given data using `page` as the backing store. The page is
     /// released back to the previous owner when the last reference is dropped using `page_tracker`.
-    pub fn new_with(data: T, page: Page<InternalClean>, page_tracker: PageTracker) -> Self {
-        let ptr = page.addr().bits() as *mut PageArcInner<T>;
+    pub fn new_with(
```

Comment:
try_new seems like a better interface, and combining them will be allow the assert to be removed. Is there a reason to keep the asserting form `new_with`?